docs/entrypoints_reference.rst

```diff
@@ -0,0 +1,172 @@
+                             optional arguments:
+                                  -h, --help            show this help message and exit
+
+                                  -host CGW-hostname, --CGW_hostname CGW-hostname
+                                                        Hostname of the server
+
+                                  -u CGW-username, --CGW_username CGW-username
+                                                        Username of Content Gateway
+
+                                  -p CGW-password, --CGW_password CGW-password
+                                                        Password for Content Gateway
+
+                                  -f CGW-filepath, --CGW_filepath CGW-filepath
+                                                        File path to read metadata
+
+``--host`` or ``--CGW_hostname``
+  Hostname of the server
+
+``-u`` or ``--CGW_username``
+  Username of Content Gateway
+
+``-P`` or ``--CGW_password``
+  Password for Content Gateway
+
+``-f`` or ``--CGW_filepath``
+  File path to read metadata
+
+
+
+Example
+***********************
+
+A typical invocation to push a metadata would look like this:
+
+.. code-block::
+
+  push-cgw-metadata \
+    --CGW_hostname https://content-gateway/example.com \
+    --CGW_username admin \
+    --CGW_password XXXXX \
+    --CGW_filepath docs/cgw/cgw_push.yaml
+
+  or
+
+  push-cgw-metadata -host https://content-gateway/example.com \
+                    -p XXXXX \
+                    -u admin \
+                    -f full_path_to_yaml_file/cgw_push.yaml
+
+
+YAML file formats
+***********************
+One can Add, Update and Delete the contents from CGW.
+The YAML file can consist details of Products, Versions and Files.
+
+The user can pass a single or multiple records of product, version and file in a single YAML file.
+
+Product
+---------------
+The highest in the hierarchy is a product. It contains significant details like `name`, `code`, `downloadpage` etc. (see below in YAML format)
+
+Version
+---------------
+Each product can have multiple product versions. Product version groups files based on when they were released. Product version is identified by its name. The version name must be unique within the product scope, however there can be other product versions with the same name in a different product scope.
+
+Besides a name the product version contains release date field, that is used in ordering product versions. If you set future date, the files will not appear in download tables until the release date.  It is also possible to add information whether the product version is the final version or early access.
+
+If you mark the product version as "hidden", all the files in the version will not be displayed and it will not be possible to download them. If the product version is marked as invisible, the files can be downloaded if someone knows the right URL, but the links will not appear in the download tables.
```

Comment:
I think this applies also to product doesn't it?

Reply:
As far as I know, this only applies to versions and their files. The product doesn’t have a release date, hidden or invisible fields.